File: ria/java/gof/src/main/java/com/gof/pattern/factoryMethod/example/OrderClient.java
package com.gof.pattern.factoryMethod.example;

public class OrderClient {
    public static void main(String[] args) {
        OrderProcessor online = OrderProcessorFactory.createOrderProcessor("online");
        online.order();

        OrderProcessor inStore = OrderProcessorFactory.createOrderProcessor("inStore");
        inStore.order();

        OrderProcessor phone = OrderProcessorFactory.createOrderProcessor("phone");
        phone.order();

        OrderProcessor refund = OrderProcessorFactory.createOrderProcessor("refund");
        refund.order();
    }
}

File: ria/gof/src/main/kotlin/com/gof/pattern/example/OrderClient.kt
package com.gof.pattern.example

fun main() {
    val online = OrderProcessorFactory.createOrderProcessor(OrderType.ONLINE)
    online.order()

    val offline = OrderProcessorFactory.createOrderProcessor(OrderType.OFFLINE)
    offline.order()

    val phone = OrderProcessorFactory.createOrderProcessor(OrderType.PHONE)
    phone.order()
}
